sgl-kernel/csrc/cpu/rope.cpp

```diff
@@ -65,8 +65,230 @@ void rope_kernel_impl(
     }
   });
 }
+
+template <typename scalar_t>
+void rotary_embedding_origin_impl(
```

Comment:
can we merge this function with the existing one. say pass a lambda as an argument.